mcp/src/main/java/io/modelcontextprotocol/spec/AsyncCloseable.java

```diff
@@ -0,0 +1,24 @@
+package io.modelcontextprotocol.spec;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * Interface for close operations that are asynchronous.
+ */
+public interface AsyncCloseable {
+
+	/**
+	 * Begins the process of closing the resource gracefully, if there is one in progress,
+	 * return the existing one.
+	 * @return A {@link Mono} that completes when the resource has been closed.
+	 */
+	Mono<Void> closeGracefully();
+
+	/**
+	 * Immediately closes the resource gracefully.
```

Comment:
The graceful close is not immediate, it kicks off asynchronously. In the cases of `void` return type it is a disconnected chain.